src/wechaty.ts

```diff
@@ -521,6 +568,51 @@ export class Wechaty extends EventEmitter implements Sayable {
     return this
   }
 
+  /**
+   * @param   {WechatyPlugin | WechatyPlugin[]} plugins      - The plugins you want to use
+   *
+   * @return  {Wechaty}                                      - this for chaining,
+   *
+   * @desc
+   * For wechaty ecosystem, allow user to define a 3rd party plugin for the current wechaty instance.
+   *
+   * @example
+   * // Report all chat message to my server.
+   *
+   * Class WechatyReportPlugin {
+   *   constructor(options) {
+   *     this.options = options;
+   *   }
+   *   install(bot) {
+   *     bot.on('message', message => http.post(this.options.url, { data: message }))
+   *   }
+   * }
+   *
+   * bot.use(new WechatyReportPlugin({ url: 'http://somewhere.to.report.your.data.com' });
+   */
+  public use (plugins: WechatyPlugin | WechatyPlugin[]) {
+    ([] as WechatyPlugin[]).concat(plugins).forEach(plugin => {
+      if (!this.plugins.includes(plugin)) {
+        this.plugins.push(plugin)
+        this.installPlugin(plugin)
+      }
+    })
+    return this
+  }
+
+  private installGloablPlugin () {
```

Comment:
Looking forward to learning from you that why do we need to install global plugins.

I feel that it can be archived for instance by instance as well.

Reply:
if you have two bots, then you have to install for every instance.
in this case, you can just install it as a global plugin.


like what it did in Vue
you can add filter in both `Vue.filter()` or `new Vue({filter: {}})`

Reply:
Got it, and I'm very happy to know that it seems you want to create lots of Wechaty instances and will manage them with your plugins!

If VUE has this kind of design, I agree that we can try this design because that's a lot of front developers who will very familiar with that.

Please go ahead with the `installGlobalPlugin()` and implement it.